deepliif/data/__init__.py

```diff
@@ -74,11 +77,33 @@ def __init__(self, dataset, batch_size, serial_batches, num_threads, max_dataset
         self.max_dataset_size = max_dataset_size
         self.dataset = dataset
         print("dataset [%s] was created" % type(self.dataset).__name__)
+
+        sampler = None
+        if os.getenv('LOCAL_RANK') is not None or os.getenv('RANK') is not None:
+            sampler = DistributedSampler(self.dataset) if len(gpu_ids) > 0 else None
+
+        # control randomness: https://pytorch.org/docs/stable/notes/randomness.html#dataloader
+        import numpy as np
+        import random
+        def seed_worker(worker_id):
+            worker_seed = torch.initial_seed() % 2**32
+            np.random.seed(worker_seed)
+            random.seed(worker_seed)
+
+        g = torch.Generator()
+        g.manual_seed(0)
+
         self.dataloader = torch.utils.data.DataLoader(
```

Comment:
Maybe to ensure we're not adding unintended effects in the DataLoader, g (generator) and sampler should be None if this is ran in the traditional setting. Does it make sense?

Reply:
@ricdodds yes you are right, let me put a conditional check for DEEPLIIF_SEED here to determine whether we use this g when we define self.dataloader (same for the seed_worker function)

Reply:
fixed